src/api.ts

```diff
@@ -36,19 +37,29 @@ export async function synchronize(
 ): Promise<OperationResult[]> {
   log('Synchronize');
 
-  const params: SynchronizeParams = {
+  let params: SynchronizeParams = {
     force: false,
     'skip-migration': false,
     ...options
   };
   const { onStarted: _, ...invokeParams } = params;
 
+  const migrationPath = getMigrationPath(config);
+  const dirExist = await existDirectory(migrationPath);
```

Comment:
Pass this value in options. Instead of double-checking inside `resolveMigrationContext`